plugins/codemirror/dialogs/codemirrordialog.js

```diff
@@ -40,20 +40,102 @@ CKEDITOR.dialog.add('codemirrordialog', function(editor) {
 			);
 
 			var preview = dialog.getContentElement('main', 'preview').getElement();
-			preview.setHtml(codeMirrorEditor.getValue());
 			preview.setSize('width', defaultWidth);
 
+			var iframe = this._createContentIframe(preview);
+			var iframeBody = iframe.$.contentDocument.body;
+
+			iframeBody.innerHTML = codeMirrorEditor.getValue();
+
 			codeMirrorEditor.on(
 				'change',
 				this._handleCodeMirrorChange.bind(this)
 			);
 		},
 
+		_createContentIframe: function(parentElement) {
+			var dialog = this.dialog;
+			var editor = dialog.getParentEditor();
+
+			var mainElement = dialog.getContentElement('main').getElement();
+
+			var tabPanel = mainElement.getAscendant(function(el) {
+				return el.getName() === 'div' && el.getAttribute('role') === 'tabpanel';
+			});
+
+			var tabPanelParent = tabPanel.getParent();
+
+			var padding = {
+				bottom: parseInt(tabPanelParent.getComputedStyle('padding-bottom'), 10) || 0,
+				top: parseInt(tabPanelParent.getComputedStyle('padding-top'), 10) || 0
+			};
+
+			var height = tabPanel.getSize('height') - (padding.bottom + padding.top);
+
+			var iframe = new CKEDITOR.dom.element('iframe');
+			parentElement.append(iframe);
+
+			iframe.setAttributes({
+				'class': 'cke_wysiwyg_frame cke_reset',
+				'frameborder': 0,
+			});
+
+			iframe.setStyles({
+				'height': height + 'px',
+				'width': '99%'
+			});
+
+			var iframeDocument = iframe.$.contentDocument;
+			var iframeHead = iframeDocument.head;
+
+			var contentsCss = editor.config.contentsCss;
+
+			if (Array.isArray(contentsCss)) {
+				contentsCss.forEach(function(url) {
+					var link = iframeDocument.createElement('link');
+					link.setAttribute('href', url);
+					link.setAttribute('rel', 'stylesheet');
+
+					iframeHead.appendChild(link);
+				});
+			}
+			else {
+				var link = iframeDocument.createElement('link');
+				link.setAttribute('href', contentsCss);
+				link.setAttribute('rel', 'stylesheet');
+
+				iframeHead.appendChild(link);
+			}
+
+			var direction = editor.config.contentsLangDirection;
+
+			var iframeHtml = iframeDocument.documentElement;
+			iframeHtml.setAttribute('dir', direction);
+			iframeHtml.setAttribute('lang', editor.config.defaultLanguage);
+
+			var iframeBody = iframeDocument.body;
+			iframeBody.classList.add('cke_editable');
+			iframeBody.classList.add('cke_editable_themed');
+			iframeBody.classList.add('cke_contents_' + direction);
+
+			iframeBody.setAttribute('contenteditable', false);
+			iframeBody.setAttribute('spellcheck', false);
+
+			iframeBody.style.background = '#fff';
```

Comment:
Is this needed? Isn't this coming from the `contentCss`?

Reply:
Yes, that's a good point.

In a "vanilla" environment, i.e. when using our build of `liferay-ckeditor` in html page that doesn't have any customization (no custom "editor" configuration) the `background-color` of the `body` inside the `iframe` is not set; but when using it in `liferay-portal` the background color was set to light grey and I'm thinking that it might be related to what you mentioned in [this](https://github.com/liferay/liferay-ckeditor/pull/122#discussion_r478861646) previous comment, so I'm also going to try to remove this and see if that fixes it.

Reply:
AFAIK `cke_contents` class adds a grey background, that's part of the `moono-lexicon` skin, so if you're adding `contentCss` files you're adding (in portal) `moono-lexicon` too, so this is needed to get rid of that style. 

Other option could be add a class `cke_preview` and overwrite that background from a css in the skin, but the result would be the same

Reply:
@carloslancha yes that's right.

So if I remove all those classes from the `iframe` and it's `body` this is what it looks like 

![no-classes-in-iframe-or-body](https://user-images.githubusercontent.com/5572/91533994-1983b800-e911-11ea-8387-fc9fb28ce768.png)

So I don't really think the "padding" is a big issue but the background does add more noise.

> Other option could be add a class cke_preview and overwrite that background from a css in the skin, but the result would be the same

I tried that too, but I was unable to make it work because another class is adding the background color to the body, and even with `background: #fff !important` it wasn't getting applied. I agree that the setting the background via "style" is a bit ugly but it works

Reply:
ok, let's keep it as it is for now and ship it... it's already better than what we have :)

Reply:
yep, you probably need to be more specific in the class `.cke_dialog.cke_preview .cke_contents .cke_editable` and `!important` or something like that, but as Chema said, is not a big issue 😄